201-vmss-linux-customimage-autoscale/azuredeploy.json

```diff
@@ -0,0 +1,449 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmSku": {
+      "type": "string",
+      "defaultValue": "Standard_D1",
+      "metadata": {
+        "description": "Size of VMs in the VM Scale Set."
+      },
+      "allowedValues": [
+        "Standard_D1",
+        "Standard_DS1",
+        "Standard_D2",
+        "Standard_DS2",
+        "Standard_D3",
+        "Standard_DS3",
+        "Standard_D4",
+        "Standard_DS4",
+        "Standard_D11",
+        "Standard_DS11",
+        "Standard_D12",
+        "Standard_DS12",
+        "Standard_D13",
+        "Standard_DS13",
+        "Standard_D14",
+        "Standard_DS14"
+      ]
+    },
+    "vmssName": {
+      "type": "string",
+      "metadata": {
+        "description": "String used as a base for naming resources. Must be 3-61 characters in length and globally unique across Azure. A hash is prepended to this string for some resources, and resource-specific information is appended."
+      },
+      "minLength": 3,
+      "maxLength": 61
+    },
+    "capacity": {
+      "type": "int",
+      "metadata": {
+        "description": "Number of VM instances (20 or less)."
+      },
+      "defaultValue": 2,
+      "maxValue": 20
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username on all VMs."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Admin password on all VMs. It must be at least 12 characters in length."
+      },
+      "minLength": 12
+    },
+    "sourceImageVhdUri": {
+      "type": "string",
+      "metadata": {
+        "description": "The source of the blob containing the custom image, must be in the same region of the deployment."
+      }
+    },
+    "appPackageUri": {
```

Comment:
Ok, I changed the appPackageUri by appPackage that should be in the /app folder as an artifact, I've tested it with the Deploy-AzureResourceGroup.ps1 with a parameters file like this one and worked:
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "vmssName": {
            "value": "jmtest034"
        },
        "adminUsername": {
            "value": "jmservera"
        },
        "capacity": {
            "value": 2
        },
        "adminPassword": {
            "value": "Th1s1sTh3P@ssw0rd"
        },
        "sourceImageVhdUri":{
            "value":"https://jmnodesrvdisks870.blob.core.windows.net/system/Microsoft.Compute/Images/vhds/jmnodeserver-osDisk.cc57607c-1824-4177-8236-df5a92f917c8.vhd"
        },
        "destinationFolder":{
            "value": "/nodeserver"
        },
        "serviceName":{
            "value": "mainsite.service"
        },
        "customScriptName":{
            "value": "updateapp.sh"
        },
        "appPackage":{
            "value": "package.tar.gz"
        }
    }
}